conda_index/index/__init__.py

```diff
@@ -124,7 +124,7 @@ def update_index(
     check_md5=False,
     channel_name=None,
     patch_generator=None,
-    threads=MAX_THREADS_DEFAULT,
+    threads: (int | None) = MAX_THREADS_DEFAULT,
```

Comment:
Is there a `from future` import to enable this syntax from python 3.9?  If so, I would recommend adding it.

Reply:
It's from future import annotations and it has been added.